File: 0001_0100/0005/0005.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    arr = [int(x) for x in f.read().split()]
    a = [x for x in arr[1:] if x & 1]
    q.write(' '.join((str(x) for x in a)) + '\n' + ' '.join((str(x) for x in arr[1:] if not x & 1)) + '\n')
    q.write("NO" if 2 * len(a) > arr[0] else "YES")

File: 0001_0100/0005/0005.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, i, odd = 0, b[100];

	fscanf(f, "%d", &n);
	for (i = 0; i < n; ++i)
		fscanf(f, "%d", b + i);
	for (i = 0; i < n; ++i)
		if (b[i] % 2)
			fprintf(q, "%d ", b[i]), ++odd;
	putc('\n', q);
	for(i = 0; i < n; ++i)
		if (!(b[i] % 2))
			fprintf(q, "%d ", b[i]);
	putc('\n', q);
	fprintf(q, 2 * odd > n ? "NO" : "YES");
	return 0;
}
